service/service.go

```diff
@@ -59,24 +49,37 @@ func Run(ctx context.Context, taxonomyRedirects map[string]string) error {
 		}},
 	})
 
+	// Create router with middleware and routes
 	router := pat.New()
 	alice := alice.New(
-		timeout.Handler(10*time.Second),
-		deprecatedLog.Handler,
+		timeoutHandler(10*time.Second),
+		log.Middleware,
 		requestID.Handler(16),
 	).Then(router)
-	routes.Setup(router, cfg)
+	routes.Setup(router, cfg, hc)
 
+	// Create HTTP server
 	server := &http.Server{
 		Addr:         cfg.BindAddr,
 		Handler:      alice,
 		ReadTimeout:  5 * time.Second,
 		WriteTimeout: 10 * time.Second,
 	}
 
-	if err = server.ListenAndServe(); err != nil {
-		return errors.Wrap(err, "failed to start server")
+	// Start the HTTP server in a new go routine
+	go func() {
+		if err := server.ListenAndServe(); err != nil {
+			log.Event(ctx, "error starting http server", log.FATAL, log.Error(err))
+			os.Exit(1)
```

Comment:
Should we exit here? In other services I think we just logged a message?

Reply:
yes, thanks!